Passing an object to `createLog` adds it to  a `data` key to the log entry

marcoalbarelli opened the pull request:
This is a proposed solution for #17 

I'm _really_ new to using tape so I might have overlloked something. If that is the case let me know

Tests are green apart from one that fails this way on master:
```
    ✖ test exited without ending
    -----------------------------
      operator: fail




  Failed Tests: There was 1 failure

    queue-change: Event Summary

      ✖ test exited without ending```

grantcarthew commented:
Hi @marcoalbarelli, this is a great idea. I didn't think about making the createLog the same as the recent changes made to addLog.

Two point:

*   I am using a simple assertion module. For consistency could you please change the string and object tests to using `is.string()` and `is.object()`. Module here: https://github.com/grantcarthew/node-rethinkdb-job-queue/blob/master/src/is.js

*   I would change the argument name from `message` to something more generic like `logValue` or `logData`, or just `data`. 

If you are really keen you could abstract the `createLog` into its own module. I have tried to keep the logic in the Queue and Job objects to a minimum for ease of maintenance. This is a big change though because you would need to add a `job-create-log.js` module and a `job-create-log.spec.js` test module. If you want to try this, there is a `test-template.js` file for test boilerplate. Although a create log module would not need the Promise support as heavily. See the `is.spec.js` for an example.

You don't need to worry about all this though. 

With respect to the tests. I ran out of time last night and left the tests in a poor state. I should be working in a branch, I know. I will fix the tests today. I am working on the `queue-change` module.

Thanks for the PR.

grantcarthew commented on src/job.js:
```diff
@@ -127,14 +127,21 @@ class Job {
-      message: message
+      message: null
     }
+    if(typeof message == 'string'){
```
is.string(message)

TomKaltz commented:
Yes!  This is what I meant with #9.  I realize now passing simple object just adds that simple object to log without all the other metadata that createLog produces.

grantcarthew commented:
It's still adding the extra log data Tom, but within the addLog call.

TomKaltz commented:
Ah yes sorry.  I was thinking of addLog.

marcoalbarelli commented:
@grantcarthew no problem about using the `is.something()` assert
I'd really like to make a separate `create-log` module too, but I'm quite in the middle of a couple of very demanding projects, so I'm not sure I ahve the time to.
Let me check if I understand how the boilerplate test works and if I have enough time to spare to make a separate module

grantcarthew commented:
@marcoalbarelli I have added a new enum I think we should use in this case. I'm using it will failed jobs now.

`seeLogData: 'See the data property attached to this log entry'`

Usage being:

`enums.message.seeLogData`

If the passed value is an object and gets assigned to the `log.data` property, set the `log.message` to the `seeLogData` value.

marcoalbarelli commented:
@grantcarthew ok on using the enum
did you push it already?
Or do you want me to add it to the enums in this PR?

grantcarthew commented:
It is in master now.

marcoalbarelli commented:
Changed code to use `is.something()`
Fixed tests description to be more talkative
Added the default message from enums if passing data as payload

Too bad I don't have time now to extract a `create-log` module, sry :|

grantcarthew commented:
Great work @marcoalbarelli 
Thanks.
